### PR TITLE
workflow: add -w option to list command to filter the workflows retrieved

### PR DESCRIPTION
**Pull request by marcdiazsan**
Closes reanahub/reana-client#452

### CODE REVIEW

**Review by audrium — COMMENTED** [review]
Looks good, please note that this PR requires rebase and the tests are failing

**Comment by marcdiazsan** [comment]
This tests fail in every case due to `get_workflows does not have parameter workflow_id_or_name` in [r-server.api](https://github.com/reanahub/reana-client/pull/559/checks?check_run_id=3658940994#step:6:81), so they wouldn't be solved when a new `reana-commons` release is made with that parameter in the API specification?

**Comment by mvidalgarcia** [comment]
Tests passing locally with [`reana-commons` PR](https://github.com/reanahub/reana-commons/pull/296) installed ✅

**Comment by mvidalgarcia** [comment]
Could you please add a new entry in the changelog?

File: CHANGES.rst [hunk 1]
```diff
@@ -4,6 +4,8 @@ Changes
 Version 0.8.0 (UNRELEASED)
 --------------------------
 
+- Changes ``list`` to filter and display all the the runs of the given workflow.
```
Comment (mvidalgarcia) [inline CHANGES.rst:7]:
```suggestion
- Adds the possibility to filter and display all the runs of a given workflow to the ``list`` command.
```

**Review by mvidalgarcia — APPROVED** [review]

**Pull request merged**